src/js/options.js

```diff
@@ -544,7 +544,41 @@ function filterTrackingDomains(/*event*/) {
 }
 
 /**
- * Add origins to the blocked resources list on scroll.
+ * Registers handlers for tracking domain toggle controls.
+ * @param toggleElement {Object} Tracking domain html element to be registered.
+ */
+// TODO unduplicate this code? since a version of it is also in popup
+function registerToggleHandlers(toggleElement) {
+  var radios = toggleElement.children('input');
+  var value = toggleElement.children('input:checked').val();
+
+  var slider = $('<div></div>').slider({
+    min: 0,
+    max: 2,
+    value: value,
+    create: function(/*event, ui*/) {
+      toggleElement.children('.ui-slider-handle').css('margin-left', -16 * value + 'px');
```

Comment:
There might be a bug here ... Not all instances of `this` in the original version of the `switch-toggle` event handler were the same `this`. We might have run into this issue before: https://github.com/EFForg/privacybadger/issues/1197#issuecomment-283152590.

Reply:
There is a similar function in `popup.js` (also called `registerToggleHandlers`), but it doesn't take any parameters. We should probably try to keep such functions identical, or move them to a common file to eliminate duplication.

Reply:
Could you help me figure out the implications of $(this) vs this here? 

I've been trying to get this function to operate on a jQuery object so it would use a 'this' like the one in popup.js does, but the more I look at it the more I feel like changing both functions to use a parameter instead.

I see that 'this' (which you appendTo) is the DOM element which registerToggleHandlers is called on, and '$(this)' (which you get other elements out of) is a jQuery wrapper around the DOM element. 
I see there's a difference in usage when jQuery returns an array, but you always call it on a single element (either on scroll or using .each) - so I can't understand the practical difference in this case.

Reply:
The potential bug I was referring to in my comment above is something else, I think. The `switch-toggle` event registration code in options.js had a bunch of instances of `$(this)`. When you refactored that code into the `registerToggleHandlers` function, you replaced all instances of `$(this)` with `toggleElement`. However, this seems like a bug (that came up once before: https://github.com/EFForg/privacybadger/issues/1197#issuecomment-283152590) because the value of `this` in JavaScript gets determined at call time, and so there are actually several different `this`'s inside the original callback, not one. The `$(this)` inside the `create` and `stop` callbacks is a different `this` from the `this` that gets radio buttons. You could add `console.log` statements (or debugger breakpoints) to see which elements are actually being referenced by the different `this` references. Let me know if this makes sense or not.

Reply:
I can write about the differences between `this` and `$(this)`; let me know if you still want me to though. It might become clear after debugging the code for a bit.

Reply:
Aaah, I see, finally! Thanks. I've fixed this and explained in the [new commit](https://github.com/EFForg/privacybadger/pull/1869/commits/a1380a0dc3d077fef6e7de5e42e0b9d6c2079635). 
Do you think I should revert to using "this" in options.js or use a parameter in popup.js?